boa_cli/src/helper.rs

```diff
@@ -124,10 +134,12 @@ impl Highlighter for LineHighlighter {
 
         let reg = Regex::new(
             r#"(?x)
-            (?P<identifier>[$A-z_]+[$A-z_0-9]*) |
+            (?P<identifier>\b[$A-z_]+[$A-z_0-9]*\b) |
```

Comment:
```suggestion
            (?P<identifier>\b[$_\p{ID_Start}][$_\p{ID_Continue}\u{200C}\u{200D}]*\b) |
```

Support more valid JS identifiers